### PR TITLE
Add more styling for ProductOverview and edge case when 0 is in stock

### LINKED ISSUE #99: Add styling for ProductOverview and add edge case for when there's 0 in stock

### PR DESCRIPTION
**Pull request by lisa-sa-li**

### CODE REVIEW

File: hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.test.js [hunk 1]
```diff
@@ -65,15 +69,37 @@ describe("<EnhancedAddToCartForm />", () => {
...
+    });
+
+    test("dropdown values are minimum between quantityAvailable and maxConstraint", () => {
+        const { queryByText, getByRole } = render(
+            <EnhancedAddToCartForm quantityAvailable={3} constraintMax={2} />
+        );
+
+        fireEvent.mouseDown(getByRole("button", { name: "1" }));
+
+        expect(queryByText("3")).not.toBeInTheDocument();
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.test.js:103]: This would still pass if the component returned nothing at all. Adding an extra check that "2" is in the document would fix that (I tested locally and that does pass).

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.js [hunk 2]
```diff
@@ -18,16 +19,34 @@ import {
...
 const Dashboard = () => {
     const reportIncident = (id) => {
         alert("Reports incident for component of id " + id);
     };
+
+    const addToCart = () => {
+        alert("add to cart");
+    };
+
+    const [mobileOpen, setMobileOpen] = React.useState(true);
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.js]: why `mobileOpen`? Is this from copying from the header component?

Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.js]: If you want to keep this around until we can actually hook it up to redux and make the info buttons open it, make the default `false`.

File: hackathon_site/dashboard/frontend/src/components/general/SideSheetRight/SideSheetRight.js [hunk 3]
```diff
@@ -8,19 +8,19 @@ import styles from "./SideSheetRight.module.scss";
 const SideSheetRight = ({ children, title, isVisible, handleClose }) => {
     return (
         <Drawer anchor="right" open={isVisible}>
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/general/SideSheetRight/SideSheetRight.js]: Can you add `onClose={handleClose}` as a prop? That way clicking outside of the drawer will also cause it to close.

File: hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.js [hunk 4]
```diff
@@ -38,40 +38,54 @@ const createQuantityList = (number) => {
...
+                <div className={styles.formButton}>
+                    <Button
+                        variant="contained"
+                        color="primary"
+                        fullWidth={true}
+                        size="large"
+                        type="submit"
+                        onClick={handleSubmit}
+                        disabled={dropdownNum === 0}
+                        data-testid="qtySelect"
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.js]: Should this test id be on the `<Select>` above?

File: hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.js [hunk 5]
```diff
@@ -38,40 +38,54 @@ const createQuantityList = (number) => {
...
         <>
             {requestFailure && (
-                <Alert className={styles.alert} variant="filled" severity="error">
+                <Alert variant="filled" severity="error">
                     {requestFailure.message}
                 </Alert>
             )}
             <form className={styles.form} onSubmit={handleSubmit}>
                 <FormControl variant="outlined" className={styles.formControl}>
                     <InputLabel>Qty</InputLabel>
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.js]: In order to be used properly with `getByLabelText` (and for accessibility reasons), this should have an `id`

File: hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.js [hunk 6]
```diff
@@ -38,40 +38,54 @@ const createQuantityList = (number) => {
...
                     <Select
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.js:65]: To correspond with the id of the `InputLabel` above, this should have a `labelId` prop

File: hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.test.js [hunk 7]
```diff
@@ -65,15 +69,37 @@ describe("<EnhancedAddToCartForm />", () => {
...
         expect(getByText(requestFailure.message)).toBeInTheDocument();
     });
+
+    test("button and select are disabled if quantityAvailable is 0", () => {
+        const { getByText, getByTestId } = render(
+            <EnhancedAddToCartForm quantityAvailable={0} />
+        );
+
+        const button = getByText("Add to cart");
+        const select = getByTestId("qtySelect");
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.test.js]: Because the test id is on the wrong element, this is actually giving you the button.

Once you add a label to the select and input label, you can use `getByLabelText("Qty")`. However since this isn't an actual html input (under the hood, it isn't actually a `<select>`), you won't be able to use `.toBeDisabled()`. You could check for the `aria-disabled="true"` tag, or the `.Mui-disabled` class.

File: hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.js [hunk 8]
```diff
@@ -115,61 +131,44 @@ const DetailInfoSection = ({
...
+                rel="noopener"
+                target="_blank"
+                startIcon={<LaunchIcon />}
+            >
+                Link
+            </Button>
+            <Typography variant="body2" className={styles.heading}>
+                Notes
+            </Typography>
+            <Typography>{notes}</Typography>
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.js]: Since the notes can have new lines in them, you could make this 

```javascript
{notes.split("\n").map((note, i) => (
     <Typography key={i}>{note}</Typography>
))}
```

otherwise it won't listen to the `\n` characters.

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.js [hunk 9]
```diff
@@ -74,6 +93,10 @@ const Dashboard = () => {
                     reportIncident={reportIncident}
                 />
                 <ReturnedTable items={itemsReturned} />
+
+                <Button variant="contained" onClick={toggleMenu}>
```
Comment (grahamhoyes) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.js:97]: Make sure we remove this eventually, but it's fine to keep in here for now until we have it hooked up to redux

**Review by grahamhoyes — APPROVED** [review]
